### ISSUE TITLE
Improvement: prevent incomplete module.properties when using some custom props

### ISSUE THREAD

**Issue by RVanhuysseXenit** [xenit-eu/alfresco-gradle-sdk#56]
#Env
gradlew 6.8.3
eu.xenit.amp 1.0.1

#Problem
When using custom values for some mandatory module properties in `sourceSet.main.amp.module`, the default values of unset properties are discarded, resulting in an incomplete module.properties file. This problem only becomes visible when trying to install the amp on an alfresco system, with the given exception 'file is not a valid amp'

#Possible solutions
* merge the default and custom properties before writing to file
* display warning when customizing module.properties but not all mandatory properties are set.

Personally i am more in favor of the 2nd solution, as it forces developers to evaluate all mandatory values (possibly preventing bad/malformed artifacts), but either would be good.

**Comment by vierbergenlars** [comment]
@RVanhuysseXenit With "display warning", you mean throw an exception and fail the build?

**Comment by RVanhuysseXenit** [comment]
> @RVanhuysseXenit With "display warning", you mean throw an exception and fail the build?

Yes that seems appropriate, since resulting artifacts do not work as they are build with the incomplete config.
